### PR TITLE
chore: add and configure mailhog for lando

### PR DESCRIPTION
**Pull request by wreality**
* Add to Laravel config to detect MailHog and change from log to SMTP configuration for mail
* Add Lando configuration for running a MailHog container.
* Update the docs with info on the `lando extras` tooling command.

This pull request also adds a new Lando tooling command, `lando extras` to manage the local "ancillary" containers we're starting to accrue.  These containers are for things that we sometimes use, like majestic or cypress, but that including in the lando config and spinning up every time would make the developer experience poopy using up their machine's resources running containers that they may not need at the moment.  

Testing the MailHog setup:
1. Run `lando extras enable mailhog`
2. Run `lando rebuild`
3. Open the MailHog interface: http://mailhog.ccr.lndo.site/
4. Run `lando artisan tinker` and paste/run the following:
```
Mail::send('welcome', [], function($message) {
    $message->to('myawesomeemail@domain.com')->from("test@example.com")->subject('Testing mails');
});
```
5. Return to http://mailhog.ccr.lndo.site and confirm that the test email has arrived in the interface.

Cleaning up:
1. Run `lando extras disable mailhog` or `lando extras all off`
2. Run `lando rebuild` to remove the MailHog containers.

### CODE REVIEW

File: scripts/lando-extras.js [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+        Object.keys(this.extras[extra].template).forEach((tKey) => {
+            if (!this.localConfig[tKey]) {
+                this.localConfig[tKey] = {};
+            }
+            Object.assign(
+                this.localConfig[tKey],
+                this.extras[extra].template[tKey]
+            );
+            this.extras[extra].enabled = true;
+        });
```
Comment (wreality) [inline scripts/lando-extras.js:45]:
```suggestion
        });
        this.extras[extra].enabled = true;
```

**Comment by gmeben** [comment]
Excellent contribution. I'll make sure to review this tomorrow morning.

**Comment by wreality** [comment]
Thanks @gmeben. I'll have the docs to include running MailHog by tomorrow morning as well so hopefully it's complete.

**Comment by wreality** [comment]
This should be fully ready for review.  It was before, but now it's EVEN BETTER. :-)

File: docs/contributing/documentation.md [hunk 2]
```diff
@@ -20,12 +20,12 @@ Github Actions powered by [Netlify](https://netlify.com) automatically build and
 
 Any pull request that contains changes to documentation will have a preview build deployed.  These previews are a great way to share proposed documentation changes for feedback and review.  
 
-Look in the automated checks section of the pull request for the action with the text "Deploy Preview Ready!".  Clicking the "Details" link will open a new tab for the deploy preview website.
+Whenever a deploy preview build is triggered, the Netlify bot will comment on the pull request with a link.  Follow this link to view the deployed documentation preview.
```
Comment (gmeben) [inline docs/contributing/documentation.md:23]:
```suggestion
Whenever a deploy preview build is triggered, the Netlify bot will comment on the pull request with a link to view the deployed documentation preview.
```

File: docs/contributing/testing.md [hunk 3]
```diff
@@ -30,7 +30,11 @@ Each package's documentation is an excellent source of information on testing be
 The [Majestic](https://github.com/Raathigesh/majestic) GUI for Jest is a useful tool for running unit tests in a browser and watching code and tests for changes.  You can add the following config to your `.lando.local.yml` file to enable a container for Majestic.
 
 ::: warning Heads Up
- If you already have configuration `.lando.local.yml`, be sure to merge the services and proxy keys, or Lando will fail to rebuild/start.
+If you already have configuration `.lando.local.yml`, be sure to merge the services and proxy keys, or Lando will fail to rebuild/start.
+:::
+
+::: tip
+The lando extras tooling command can setup Majestic for you.  Run: `lando extras enable majestic`
```
Comment (gmeben) [inline docs/contributing/testing.md:37]: noun should be a verb
```suggestion
The lando extras tooling command can set up Majestic for you.  Run: `lando extras enable majestic`
```

File: docs/contributing/testing.md [hunk 4]
```diff
@@ -66,6 +70,8 @@ We use [Cypress](https://www.cypress.io/) for our integration testing.  Cypress
 
 ::: tip
 These instructions focus on installing Cypress under Lando.  Cypress can bit a bit of a resource hog and, as such, might be better run directly in your host environment.
+
+Also you can use Lando Extras to setup this configuration for you.  Run: `lando extras enable cypress`
```
Comment (gmeben) [inline docs/contributing/testing.md:74]: noun should be a verb
```suggestion
Also you can use Lando Extras to set up this configuration for you.  Run: `lando extras enable cypress`
```

File: scripts/lib/lando-extras.js [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+class LandoExtras {
+    #ymlPath;
+    #localConfigFile;
+    #extrasConfigFile;
+    #extras = [];
+    #localConfig = {};
+
+    /**
+     * Setup LandoExtras.
```
Comment (gmeben) [inline scripts/lib/lando-extras.js:12]: noun should be a verb
```suggestion
     * Set Up LandoExtras.
```

File: docs/index.md [hunk 6]
```diff
@@ -14,8 +14,8 @@ In here should go a nice description about CCR and what it's intended to do.
 ::: slot home-banner
 
 ::: danger HEADS UP
-  CCR is currently in the initial stages of development and isn't ready for deployment or testing just yet.  
-  Check out our  [contribution guidelines](https://github.com/MESH-Research/CCR/blob/master/.github/CONTRIBUTING.md) to help make CCR a reality.
+  CCR is currently in the initial development stage and isn't ready for deployment or testing just yet.
```
Comment (gmeben) [inline docs/index.md:17]: It's misleading to say 'CCR isn't ready for testing' when unit and integration tests have been implemented, but I respect if a different meaning was intended and there's a preference to change this to something else or preserve it.
```suggestion
  CCR is currently in the initial development stage and isn't ready for deployment just yet.  
```

File: scripts/extras.js [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+ *
+ * @param {LandoExtras} extras
+ */
+function getActionsList(extras) {
+    const list = extras.map((e) => {
+        const { name, description, enabled } = e;
+        return {
+            name,
+            description,
+            icon: enabled ? `[${success("✔")}]` : "[ ]",
```
Comment (gmeben) [inline scripts/extras.js:78]: Love this

File: scripts/lib/lando-extras.js [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+     */
+    disableAll() {
+        this.#extras.forEach((extra) => {
+            this.#disableTemplate(extra.template);
+            extra.enabled = false;
+        });
+    }
+
+    /**
+     * Enable an extra by name.
```
Comment (gmeben) [inline scripts/lib/lando-extras.js:147]:
```suggestion
     * Disable an extra by name.
```

File: scripts/lib/lando-extras.js [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+            Object.keys(tContent).forEach((key) => {
+                delete this.#localConfig[tKey][key];
+            });
+        });
+    }
+
+    /**
+     * Return true if a config template is enabled in localConfig
+     *
+     * @param {Object} config Configuration template to check.
```
Comment (gmeben) [inline scripts/lib/lando-extras.js:94]:
```suggestion
     * @param {Object} template Configuration template to check.
```

File: scripts/lib/lando-extras.js [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+        const extra = this.#get(name);
+        this.#enableTemplate(extra.template);
+        extra.enabled = true;
+    }
+
+    /**
+     * Map extra configs.
+     *
+     * @param Callable callback
+     * @returns {any} Results of map call.
```
Comment (gmeben) [inline scripts/lib/lando-extras.js:172]:
```suggestion
     * @returns {any} callback Results of map call.
```

**Review by gmeben — CHANGES_REQUESTED** [review]
Looks very good overall. I only noticed minor things to be changed.

**Comment by gmeben** [comment]
I forgot to mention that Laravel Tinker did not appreciate the whitespace within the code included in the instructions, so I ended up running this instead for it to succeed:
```
Mail::send('welcome', [], function($message) { $message->to('myawesomeemail@domain.com')->from("test@example.com")->subject('Testing mails'); });
```

File: docs/index.md (hunk 6 above)
Comment (wreality) [reply docs/index.md:17]:
```suggestion
  CCR is currently in the initial development stage and isn't ready for deployment or evaluation just yet.  
```

File: scripts/lib/lando-extras.js [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+    disable(name) {
+        const extra = this.#get(name);
+        this.#disableTemplate(extra.template);
+        extra.enabled = false;
+    }
+
+    /**
+     * Enable an extra by name.
+     *
+     * @param {string} name Name of extra to disable
```
Comment (wreality) [inline scripts/lib/lando-extras.js:160]:
```suggestion
     * @param {string} name Name of extra to enable
```

File: scripts/lib/lando-extras.js [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+        this.#extras.forEach((extra) => {
+            this.#disableTemplate(extra.template);
+            extra.enabled = false;
+        });
+    }
+
+    /**
+     * Enable an extra by name.
+     *
+     * @param {string} name Name of extra to enable
```
Comment (wreality) [inline scripts/lib/lando-extras.js:149]:
```suggestion
     * @param {string} name Name of extra to disable
```

File: docs/index.md (hunk 6 above)
Comment (gmeben) [reply docs/index.md:17]: That works

File: scripts/lib/lando-extras.js [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+    enable(name) {
+        const extra = this.#get(name);
+        this.#enableTemplate(extra.template);
+        extra.enabled = true;
+    }
+
+    /**
+     * Map extra configs.
+     *
+     * @param Callable callback
```
Comment (wreality) [inline scripts/lib/lando-extras.js:171]:
```suggestion
     * @param {Function} callback
```

File: scripts/lib/lando-extras.js (hunk 10 above)
Comment (wreality) [reply scripts/lib/lando-extras.js:172]: This is an @returns so there isn't a name (and the function doesn't return the callback but rather the result of the user-supplied callback.

**Comment by wreality** [comment]
Changes incorporated.

**Review by gmeben — APPROVED** [review]

**Pull request merged**